client_plugin/drivers/shared/dockerops/dockerops.go

```diff
@@ -65,14 +66,44 @@ const (
 	checkDuration = 5 * time.Second
 	// Time between successive checks for deleting a volume
 	checkSleepDuration = time.Second
-	// Timeout to mark Samba service launch as unsuccessful
-	sambaRequestTimeout = 30 * time.Second
+	// default Timeout to mark Samba service launch as unsuccessful
+	defaultSambaRequestTimeOutInSecond = 30
 	// Prefix for internal volume names
 	internalVolumePrefix = "InternalVol"
 	// Error returned when no Samba service for that volume exists
 	noSambaServiceError = "No file service exists"
 )
 
+var (
+	SambaRequestTimeout      time.Duration
+	isSambaRequestTimeoutSet bool = false
+)
+
+// return SambaRequest timeout value
+func GetSambaRequestTimeout() time.Duration {
+	if isSambaRequestTimeoutSet != false {
+		return SambaRequestTimeout
+	}
+	// read sambaRequest Timout value from enviroment variable
+	timeOutFromConfig := os.Getenv("VFILE_TIMEOUT_IN_SECOND")
```

Comment:
I do not think there is a need for `isSambaRequestTimeoutSet `.
This should make do, won't it ? 
 
```go
func GetSambaRequestTimeout() time.Duration {
 timeOutSec, err := strconv.Atoi(os.Getenv("VFILE_TIMEOUT_IN_SECOND"))
 if err != nil {
   timeOutSec = defaultSambaRequestTimeOutInSecond 
 }
  log.WithFields(log.Fields{"value": timeOutSec ,}).Info("sambaRequest timeout")
  return time.Duration(timeOutSec ) * time.Second
}
```

  
Also, this hjas nothing to do with Samba, it is a generic timeout for service start. Please rename accordingly